frontend/app/services/subscription-service.server.ts

```diff
@@ -212,6 +267,62 @@ function createSubscriptionService() {
     }
   }
 
+  async function createSubscription(userId: string, subscription: SubscriptionInfo) {
+    const auditService = getAuditService();
+    const instrumentationService = getInstrumentationService();
+    auditService.audit('alert-subscription.update', { userId });
+    const url = new URL(`${CDCP_API_BASE_URI}/api/v1/users?raoidcUserId=${userId}`);
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    instrumentationService.countHttpStatus('http.client.cdcp-api.users.gets', response.status);
+
+    if (!response.ok) {
+      log.error('%j', {
+        message: 'Failed find data',
+        status: response.status,
+        statusText: response.statusText,
+        url: url,
+        responseBody: await response.text(),
+      });
+
+      throw new Error(`Failed to find data. Status: ${response.status}, Status Text: ${response.statusText}`);
+    }
+
+    const users = await response.json();
+    if (users._embedded.users.length === 0) {
+      throw new Error(`Failed to find the user: ${userId}.`);
+    }
+    const userParsed = userSchema.parse(users._embedded.users[0]);
+
+    const subscriptionsURL = userParsed._links.subscriptions.href;
+    const subscriptionsResponse = await fetch(subscriptionsURL, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(subscription),
+    });
+
+    instrumentationService.countHttpStatus('http.client.cdcp-api.subscriptions.create', subscriptionsResponse.status);
+
+    if (!subscriptionsResponse.ok) {
+      log.error('%j', {
+        message: 'Failed to create data',
+        status: subscriptionsResponse.status,
+        statusText: subscriptionsResponse.statusText,
+        url: url,
+        responseBody: await subscriptionsResponse.text(),
+      });
+
+      throw new Error(`Failed to create data. Status: ${subscriptionsResponse.status}, Status Text: ${subscriptionsResponse.statusText}`);
+    }
+  }
```

Comment:
I think it might be beneficial to return the new subscription object to the caller, since it will contain additional data like the ID.

If the API doesn't return the subscription object upon creation, then it should be fixed.